os/include/semaphore.h

```diff
@@ -79,13 +79,19 @@
 #define PRIOINHERIT_FLAGS_DISABLE (1 << 0) /* Bit 0: Priority inheritance
 					    * is disabled for this semaphore */
 #define FLAGS_INITIALIZED         (1 << 1) /* Bit 1: This semaphore initialized */
+#define FLAGS_SIGSEM              (1 << 2) /* Bit 2: The semaphore for signaling */
```

Comment:
Setting this flag is only when `#if defined(CONFIG_BINMGR_RECOVERY) && defined(__KERNEL__)`